src/utils/daos.c

```diff
@@ -0,0 +1,870 @@
+	uuid_clear(ap->pool_uuid);
+	uuid_clear(ap->cont_uuid);
+	ap->mdsrv_str = NULL;
+	ap->mdsrv = NULL;
+	ap->attrname_str = NULL;
+	ap->value_str = NULL;
+	ap->path_str = NULL;
+	ap->type_str = NULL;
+	ap->oclass_str = NULL;
+	ap->chunk_size = 0;
+	ap->snapname_str = NULL;
+	ap->epc = 0;
+	ap->epcrange_str = NULL;
+	ap->epcrange_begin = 0;
+	ap->epcrange_end = 0;
+}
+
+
+static void
+cmd_args_print(struct cmd_args_s *ap)
+{
+	if (ap == NULL)
+		return;
+
+	D_INFO("\tgroup=%s\n", ap->group);
+	D_INFO("\tpool UUID: "DF_UUIDF"\n", DP_UUID(ap->pool_uuid));
+	D_INFO("\tcont UUID: "DF_UUIDF"\n", DP_UUID(ap->cont_uuid));
+
+	D_INFO("\tpool svc: parsed %u ranks from input %s\n",
+		ap->mdsrv ? ap->mdsrv->rl_nr : 0,
+		ap->mdsrv_str ? ap->mdsrv_str : "NULL");
+
+	D_INFO("\tattr: name=%s, value=%s\n",
+		ap->attrname_str ? ap->attrname_str : "NULL",
+		ap->value_str ? ap->value_str : "NULL");
+	D_INFO("\tpath=%s, type=%s, oclass=%s, chunk_size="DF_U64"\n",
+		ap->path_str ? ap->path_str : "NULL",
+		ap->type_str ? ap->type_str : "NULL",
+		ap->oclass_str ? ap->oclass_str : "NULL",
+		ap->chunk_size);
+	D_INFO("\tsnapshot: name=%s, epoch="DF_U64", epoch range=%s "
+		"("DF_U64"-"DF_U64")\n",
+		ap->snapname_str ? ap->snapname_str : "NULL",
+		ap->epc,
+		ap->epcrange_str ? ap->epcrange_str : "NULL",
+		ap->epcrange_begin, ap->epcrange_end);
+}
+
+static daos_size_t
+tobytes(const char *str)
+{
+	daos_size_t	 size;
+	char		*end;
+
+	size = strtoull(str, &end, 0);
+	/* Prevent negative numbers from turning into unsigned */
+	if (str != NULL && str[0] == '-') {
+		fprintf(stderr, "WARNING bytes < 0 (string %s)"
+				"converted to "DF_U64" : using 0 instead\n",
+				str, size);
+		size = 0;
+		return size;
+	}
+
+	/** no suffix used */
+	if (*end == '\0')
+		return size;
+
+	/** let's be permissive and allow MB, Mb, mb ...*/
+	if (*(end + 1) != '\0' &&
+	    ((*(end + 1) != 'b' && *(end + 1) != 'B') || (*(end + 2) != '\0')))
+		return 0;
+
+	switch (*end) {
+	case 'b':
+	case 'B':
+		break;
+	case 'k':
+	case 'K':
+		size <<= 10;
+		break;
+	case 'm':
+	case 'M':
+		size <<= 20;
+		break;
+	case 'g':
+	case 'G':
+		size <<= 30;
+		break;
+	case 't':
+	case 'T':
+		size <<= 40;
+		break;
+	case 'p':
+	case 'P':
+		size <<= 50;
+		break;
+	case 'e':
+	case 'E':
+		size <<= 60;
+		break;
+	default:
+		return 0;
+	}
+
+	return size;
+}
+
+/* acopy_str()
+ * Allocate dest string and copy source contents.
+ * Return NULL if not possible. Caller ensures src is not NULL.
+ */
+static char *
+acopy_str(const char *src)
+{
+	char	*dst = NULL;
+
+	assert(src != NULL);
+
+	/* TODO: put a reasonable max on strlen(src) */
+	D_ALLOC_ARRAY(dst, strlen(src) + 1);
+	if (dst != NULL)
+		strcpy(dst, src);
+
+	return dst;
+}
+
+static int
+epoch_range_parse(struct cmd_args_s *ap)
+{
+	const char	*sep = "-";
+	char		*s, *s_saved;
+	char		*end_s;
+	char		*p;
+
+	s = s_saved = strdup(ap->epcrange_str);
+	if (s == NULL) {
+		fprintf(stderr, "strdup() failed\n");
+		goto out;
+	}
+
```

Comment:
can't you just use sscanf()?

Reply:
will attempt sscanf() this in next patch. malformed input error catching issues were considered. This also was created starting from some other strtok-based parsing function, daos_rank_list_parse()